util/runner/auto.zig

```diff
@@ -0,0 +1,137 @@
+const std = @import("std");
+const expectEq = std.testing.expectEqual;
+const expectEqSlices = std.testing.expectEqualSlices;
+
+const Runner = @import("./types.zig");
+const AutoRunner = @This();
+
+const State = union(enum) {
+    preparing: Preparing,
+    running: Running,
+
+    const Preparing = struct {
+        /// Number of iterations to be performed in the benchmark.
+        N: usize = 1,
+
+        /// Maximum number of iterations the benchmark can run. This limit helps
+        /// to avoid excessively long benchmark runs.
+        max_iterations: u16,
+
+        /// Time budget for the benchmark in nanoseconds. This value is used to
+        /// determine how long a single benchmark should be allowed to run
+        /// before concluding. Helps in avoiding long-running benchmarks.
+        time_budget_ns: u64,
+
+        iterations_remaining: usize = 1,
```

Comment:
please add comments for auto-doc for the sake of complete documentation.

Reply:
Happy to, though neither `State` nor `State.Preparing` are exported, it's all internal.